Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3458,50 +3459,177 @@ def panorama_delete_url_filter_command(url_filter_name: str):
+        'GroupTag': rule_get('group-tag'),
+        'Schedule': rule_get('schedule'),
+        'LogStart': rule_get('log-start'),
+        'LogForwardingProfile': rule_get('log-setting'),
+        'NegateSource': rule_get('negate-source'),
+    }
+    pretty_rule: Dict[str, Any] = {}
+
+    context_rule['DeviceGroup'] = DEVICE_GROUP
+
+    context_rule['Location'] = pretty_rule['Location'] = rule.get('@loc')
+
+    context_rule['SecurityProfileGroup'] = pretty_rule['Profile Group'] = rule_get('profile-setting', subkeys=('group', 'member', '#text'))
+
+    profiles = {
+        k: xml_get(pfs, k)
+        for k in pfs.keys()
+        if not str(k).startswith('@')
+    } \
+        if isinstance(pfs := dict_safe_get(rule, ('profile-setting', 'profiles')), dict )\
+        else None
+
+    pretty_rule['Profile'] = context_rule['SecurityProfile'] = profiles
+
+    target = dict_safe_get(rule, ('target',), {}, dict)
+    target |= {
+        'devices': xml_get(target, 'devices'),
+        'negate': xml_get(target, 'negate'),
+    }
+    context_rule['Target'] = target
+    pretty_rule['Target'] = (
+        f'negate: {target["negate"]}\n'
+        f'devices: {target["devices"]}'
+    ).replace('[', '').replace(']', '').replace("'", '')
+
+    context_rule['Name'] = pretty_rule['Name'] = rule_get('@name')
+    context_rule['QoSMarking'] = dict_safe_get(rule, ('qos', 'marking'))
+    context_rule['Type'] = pretty_rule['Type'] = rule_get('rule-type')
+    context_rule['From'] = pretty_rule['Source Zone'] = rule_get('from')
+    context_rule['DestinationDevice'] = pretty_rule['Destination Device'] = rule_get('destination-hip')
+    context_rule['Action'] = pretty_rule['Action'] = rule_get('action')
+    context_rule['SourceDevice'] = pretty_rule['Source Device'] = rule_get('source-hip')
+    context_rule['Tags'] = pretty_rule['Tags'] = rule_get('tag')
+    context_rule['SourceUser'] = pretty_rule['Source User'] = rule_get('source-user')
+    context_rule['Application'] = pretty_rule['Application'] = rule_get('application')
+    context_rule['Service'] = pretty_rule['Service'] = rule_get('service')
+    context_rule['To'] = pretty_rule['Destination Zone'] = rule_get('to')
+    context_rule['Source'] = pretty_rule['Source Address'] = rule_get('source')
+    context_rule['CustomUrlCategory'] = pretty_rule['Url Category'] = rule_get('category')
+    context_rule['Destination'] = pretty_rule['Destination Address'] = rule_get('destination')
+
+    context_rule['Option'] = option = rule.get('option')
+    pretty_rule['Options'] = [
+        tick_box.get(rule_get('log-start'), '') + ' Log at Session Start',
+        f'Log Forwarding: {rule_get("log-setting")}',
+        f'Schedule: {rule_get("schedule")}',
+        f'QoS Marking: {next(filter(lambda k: not k.startswith("@"), qos.keys()), None) if isinstance(qos := context_rule["QoSMarking"], dict) else qos}',  # type: ignore[assignment, arg-type, attr-defined, union-attr] # pylint: disable=assignment
```

Comment:
pretty hard to read IMO, can we make it a bit simpler?

Reply:
on it